packages/core/ios/RNEmbrace/BundleID.swift

```diff
@@ -0,0 +1,67 @@
+import Foundation
+import EmbraceIO
+import CryptoKit
+
+private struct LastBundleComputation: Codable {
+    private static let STORAGE_KEY = "EMBRACE_LAST_BUNDLE_COMPUTATION"
+    var path: String
+    var computedAt: Date?
+    var id: String
+
+    func store () {
+        if let encoded = try? JSONEncoder().encode(self) {
+            UserDefaults.standard.set(encoded, forKey: LastBundleComputation.STORAGE_KEY)
+        }
+    }
+
+    static func fromStorage() -> LastBundleComputation {
+        if let storedData = UserDefaults.standard.object(forKey: LastBundleComputation.STORAGE_KEY) as? Data,
+           let loaded = try? JSONDecoder().decode(LastBundleComputation.self, from: storedData) {
+            return loaded
+        }
+
+        return LastBundleComputation(path: "", computedAt: nil, id: "")
+    }
+}
+
+enum ComputeBundleIDErrors: Error {
+    case emptyPath
+}
+
+struct BundleID {
+    var id: String
+    var cached: Bool
+}
+
+func computeBundleID(path: String) throws -> BundleID {
+    if path.isEmpty {
+        throw ComputeBundleIDErrors.emptyPath
+    }
+
+    var last = LastBundleComputation.fromStorage()
+
+    // Check if we can re-use our last computed bundle ID, this is true only if the path hasn't changed and
+    // the file contents at that path haven't been modified since the last time we computed the ID
+    // TODO in ios 5.x and Android implementation we are just returning last.id in this case without checking modified at...is that correct?
```

Comment:
@ArielDemarco @TheMutsi this logic feels correct to me, e.g. only re-use the cache if the path is the same and the contents haven't been updated but it does deviate from the previous implementations so wanted to double-check

Reply:
LGTM; I'd wait for @TheMutsi to see if there's nothing we're not considering

Reply:
@ArielDemarco can we move it to DispatchQueue ? I think it could be very heavy

Reply:
I think that makes sense, I moved the calls to compute ID to background queues